src/socket/ip.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2025 International Digital Economy Academy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <arpa/inet.h>
+#include <string.h>
+#include <netdb.h>
+#ifdef _WIN32
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#else
+#include <sys/socket.h>
+#include <netinet/in.h>
+#endif
+#include <moonbit.h>
+
+static inline void empty_function(void *self)
+{
+    // no-op: empty function
+}
+
+MOONBIT_FFI_EXPORT
+int32_t
+moonbitlang_ipv4_addrlen(void)
+{
+    return INET_ADDRSTRLEN;
+}
+
+MOONBIT_FFI_EXPORT
+int32_t
+moonbitlang_ipv6_addrlen(void)
+{
+    return INET6_ADDRSTRLEN;
+}
+
+typedef struct moonbit_get_ip_name_port_cb_s
+{
+    int32_t (*code)(
+        struct moonbit_get_ip_name_port_cb_s *,
+        int32_t ai_family,
+        int32_t port,
+        int32_t flowinfo,
+        int32_t scope_id);
+} moonbit_get_ip_name_port_cb_t;
+MOONBIT_FFI_EXPORT
+int32_t
+moonbitlang_get_ip_addr_detail(
+    const struct sockaddr *src,
+    moonbit_get_ip_name_port_cb_t *cb)
+{
```

Comment:
I think this function is unnecessary and too complex. `ai_family`, `port`, etc. can be accessed via simple getter functions